server/src/main/java/org/apache/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -950,15 +950,11 @@ public DataSegment map(int index, ResultSet r, StatementContext ctx) throws SQLE
         }
     );
 
-    if (segments == null || segments.isEmpty()) {
-      log.info("No segments found in the database!");
+    if (segments == null) {
```

Comment:
This code is pointless, since `.list()` above cannot return null and `inReadOnlyTransaction` will not convert a nonnull return to null. Might as well remove it, or include a `Preconditions.checkNotNull` or something (throw an exception on the unanticipated `null` rather than returning early and leaving the `snapshot` null).

Reply:
`Preconditions` check makes sense to me, to control the error messaging here instead of allowing the NPE to happen downstream, will update

Reply:
Er, actually this is just in the poll, i guess the NPE would still potentially happen, regardless that approach seems cleaner.